src/plugins/intel_cpu/src/emitters/snippets/aarch64/utils.cpp

```diff
@@ -154,28 +153,52 @@ void push_and_load_ptrs_with_offsets(dnnl::impl::cpu::aarch64::jit_generator* h,
     const auto sp_size = rnd_up(mem_ptrs.size() * gpr_length, sp_alignment);
     h->sub(h->sp, h->sp, sp_size);
 
-    // Push all pointers with offsets onto stack
+    // Generate stack offsets for sequential storage
+    std::vector<size_t> stack_offsets;
+    stack_offsets.reserve(mem_ptrs.size());
+    for (size_t i = 0; i < mem_ptrs.size(); i++) {
+        stack_offsets.push_back(i * gpr_length);
+    }
+
+    // Use the common function to push pointers with offsets to stack
+    push_ptrs_with_offsets_to_stack(h, mem_ptrs, memory_offsets, buffer_ids, aux_regs, stack_offsets);
+
+    // Load back the adjusted pointers to specified registers
+    for (size_t i = 0; i < load_regs.size() && i < mem_ptrs.size(); i++) {
+        h->ldr(load_regs[i], Xbyak_aarch64::ptr(h->sp, static_cast<int32_t>(i * gpr_length)));
+    }
+
+    // Restore stack pointer
+    h->add(h->sp, h->sp, sp_size);
+}
+
+void push_ptrs_with_offsets_to_stack(dnnl::impl::cpu::aarch64::jit_generator* h,
+                                     const std::vector<Xbyak_aarch64::XReg>& mem_ptrs,
+                                     const std::vector<size_t>& memory_offsets,
+                                     const std::vector<size_t>& buffer_ids,
+                                     const std::vector<Xbyak_aarch64::XReg>& aux_regs,
+                                     const std::vector<size_t>& stack_offsets) {
+    OV_CPU_JIT_EMITTER_ASSERT(mem_ptrs.size() == memory_offsets.size(), "mem_ptrs and memory_offsets size mismatch");
+    OV_CPU_JIT_EMITTER_ASSERT(mem_ptrs.size() == buffer_ids.size(), "mem_ptrs and buffer_ids size mismatch");
+    OV_CPU_JIT_EMITTER_ASSERT(mem_ptrs.size() == stack_offsets.size(), "mem_ptrs and stack_offsets size mismatch");
+
+    // Store all pointers with offsets to their specific stack locations
     for (size_t i = 0; i < mem_ptrs.size(); i++) {
         const auto& ptr_reg = mem_ptrs[i];
-        int32_t stack_offset = i * gpr_length;
+        auto stack_offset = static_cast<int32_t>(stack_offsets[i]);
```

Comment:
If we work with `int32_t`, can we pass already casted vector and have `const std::vector<int32_t>& stack_offsets` in arguments? To avoid explicit casting in the helper

Reply:
Done